tensorflow/core/common_runtime/gpu/gpu_init.h

```diff
@@ -18,14 +18,20 @@ limitations under the License.
 
 #include "tensorflow/core/lib/core/status.h"
 
+#if GOOGLE_CUDA
+#define GPU_PLATFORM_NAME "CUDA"
+#else
+#define GPU_PLATFORM_NAME ""
```

Comment:
Instead of using an empty string, I'd rather see "NO_GPU_PLATFORM", maybe through a constant kNoGpuConstant. Then caller can return an error status indicating no GPU platform has been found.